### ISSUE TITLE
Piecewise function and matrices support

### ISSUE THREAD

**Issue by sun21170** [daniel3735928559/guppy#107]
Can I render piecewise function or matrix using this editor? Piecewise function latex would be f(x)=\begin{cases}{a} & x = 0\\{b} & x > 0\end{cases} and for matrix it would be  \begin{bmatrix}a & b\\c & d \end{bmatrix}. I could not find a way of doing this in the demo for guppy.

**Comment by JMSS-Unknown** [comment]
Matrices are already built in. Type mat and use ';' and ',' to add rows and column. To use piecewise, you will probably have to create your own symbol.

**Comment by JMSS-Unknown** [comment]
Actually, https://github.com/daniel3735928559/guppy/issues/84#issuecomment-357977609 states that there is currently no way to limit it to two columns

**Comment by sun21170** [comment]
> Matrices are already built in. Type mat and use ';' and ',' to add rows and columns. To use piecewise, you will probably have to create your own symbol.

@JMSS-Unknown , When I type mat in the standard mathquill demo, it does't show a matrix structure, but just the text I typed. So, what did you mean when you said matrices are supported?

**Comment by akrinke** [comment]
@sun21170 On https://guppy.js.org/site/ if you type `mat` it will be replaced by `( )`. Then you can enter the matrix using `;` and `,`.

**Comment by sun21170** [comment]
> @sun21170 On https://guppy.js.org/site/ if you type `mat` it will be replaced by `( )`. Then you can enter the matrix using `;` and `,`.

@akrinke , Yes, that certainly works. I was thinking this was for mathquill and tried it on mathquill.com. Sorry, my misunderstanding. Is there some way of rendering piecewise function in the guppy demo or its not possible at this time according to one of the replies in this thread?

**Comment by JMSS-Unknown** [comment]
Not on the demo, but if you put it on your own site (it can be downloaded locally) and modify symbols.json with the examples I gave then you can render piecewise

**Comment by sun21170** [comment]
> Not on the demo, but if you put it on your own site (it can be downloaded locally) and modify symbols.json with the examples I gave then you can render piecewise

@JMSS-Unknown , I will give that a try and let you know. Thanks for your great help.
